### ISSUE TITLE
Custom rules are not applied on Android

### ISSUE THREAD

**Issue by herrbert74** [arturbosch/detekt#2169]
I added detekt to both my work and hobby project, both multi module Android projects. It works perfectly.
However, if I try to add custom rules, I can't get them to work. I tried to add rules that would definitely fail to no avail.
Sometimes the detekt task just quietly succeeds without any warning, sometimes it gives the following warnings for each module (The custom ruleset is called "babe"):

Property 'babe' is misspelled or does not exist.
io.gitlab.arturbosch.detekt.cli.InvalidConfig: Run failed with 1 invalid config property.

This probably means that the detekt config file is OK, but I got something wrong with the Gradle config.

My hobby project is open source, please have a look at here in the detekt branch:
https://bitbucket.org/herrbert74/companies-house

The custom rules are in the "core-detekt" module. All configs for Gradle are in the main build.gradle file under the allprojects and subprojects configurations. Some files are under the team-props folder in the project directory.

All in all I spent two days trying everything, without any luck. Any help would be greatly appreciated.

**Comment by thuytrinh** [comment]
I've upgraded to v1.2.2 and also encountered this problem. In my case, there are some custom detekt rules defined in an own Gradle module: `DetektRules`. And this module is linked to detekt by:

```groovy
dependencies {
    detektPlugins project(":DetektRules")
}
```
```groovy
tasks.withType(io.gitlab.arturbosch.detekt.Detekt) {
    // Custom detekt rules should be built before we run `detekt`.
    // See https://arturbosch.github.io/detekt/extensions.html#pitfalls.
    dependsOn(":DetektRules:build")
}
```

In the detekt config file, the custom rules are declared like this:

```yml
custom-rules:
  AbsentOrWrongCopyrightHeader:
    active: false
```

After upgrading to v1.2.2, having that custom rule declaration will always make detekt succeed even through there's some violation. If I remove that custom rule declaration, detekt managed to report violation normally again.

**Comment by arturbosch** [comment]
Sorry for the late response. detekt now validates the configuration file and every property.
If it does not know it, it will tell you on the console which property it does not know and exists/throws.
I've added some tests in the sample project showing how to exclude custom properties here: https://github.com/arturbosch/detekt/pull/2192/files

Does it know works for you?

> After upgrading to v1.2.2, having that custom rule declaration will always make detekt succeed even through there's some violation. If I remove that custom rule declaration, detekt managed to report violation normally again.

This should not happen and seems to be the same as #2168

**Comment by herrbert74** [comment]
Thanks, that solved it.
